src/mlpack/tests/decision_tree_regressor_test.cpp

```diff
@@ -0,0 +1,1047 @@
+
+//   // Build the tree.
+//   DecisionTree<InformationGain> tree(fullData, di, fullLabels, 5, weights, 10);
+
+//   // Now evaluate the accuracy of the tree.
+//   arma::Row<size_t> predictions;
+//   tree.Classify(testData, predictions);
+
+//   REQUIRE(predictions.n_elem == testData.n_cols);
+//   size_t correct = 0;
+//   for (size_t i = 0; i < testData.n_cols; ++i)
+//     if (testLabels[i] == predictions[i])
+//       ++correct;
+
+//   // Make sure we got at least 70% accuracy.
+//   const double correctPct = double(correct) / double(testData.n_cols);
+//   REQUIRE(correctPct > 0.70);
+// }
+
+/**
+ * Test that the decision tree generalizes reasonably.
+ */
+TEST_CASE("SimpleGeneralizationTest_", "[DecisionTreeRegressorTest]")
+{
+
+  // Loading data.
+  data::DatasetInfo info;
+  arma::mat trainData, testData;
+  arma::Row<double> trainLabels, testLabels;
+  arma::rowvec weights = arma::ones<arma::rowvec>(355);
+  LoadBostonHousingDataset(trainData, testData, trainLabels, testLabels, info);
+
+  // Build decision tree.
+  DecisionTreeRegressor<MSEGain> d(trainData, info, trainLabels);
+
+  // Get the predicted test labels.
+  arma::Row<double> predictions;
+  d.Predict(testData, predictions);
+
+  REQUIRE(predictions.n_elem == testData.n_cols);
+
+  // Figure out rmse.
+  double rmse = RMSE(predictions, testLabels);
+
+  // REQUIRE(rmse < 9.21);
+  // std::cout << predictions << std::endl << testLabels;
+  arma::Row<double> trainPred;
+  d.Predict(trainData, trainPred);
+  // std::cout << trainPred;
+
+  std::cout << "Train RMSE: " << RMSE(trainLabels, trainPred) << std::endl;
+}
+
+/**
+ * Test that the decision tree generalizes reasonably when built on float data.
+ */
+// TEST_CASE("SimpleGeneralizationFMatTest_", "[DecisionTreeRegressorTest]")
+// {
+//   // Loading data.
+//   data::DatasetInfo info;
+//   arma::mat trainData, testData;
+//   arma::Row<double> trainLabels, testLabels;
+//   LoadBostonHousingDataset(trainData, testData, trainLabels, testLabels, info);
+
+//   // Initialize an all-ones weight matrix.
+//   arma::rowvec weights(trainLabels.n_cols, arma::fill::ones);
+
+//   // Build decision tree.
+//   DecisionTreeRegressor<> d(trainData, trainLabels);
+//   DecisionTreeRegressor<> wd(trainData, trainLabels, weights);
+
+//   // Get the predicted test labels.
+//   arma::Row<double> predictions;
+//   d.Predict(testData, predictions);
+
+//   REQUIRE(predictions.n_elem == testData.n_cols);
+
+//   // Figure out the rmse.
+//   double rmse = RMSE(predictions, testLabels);
+
+//   REQUIRE(rmse < 9.21);
+//   std::cout << R2Score(predictions, testLabels) << std::endl;
+
+//   // Reset the prediction.
+//   predictions.zeros();
+//   wd.Predict(testData, predictions);
+
+//   REQUIRE(predictions.n_elem == testData.n_cols);
+
+//   // Figure out the rmse.
+//   double wdrmse = RMSE(predictions, testLabels);
+
+//   REQUIRE(wdrmse < 9.21);
+// }
```

Comment:
Can you be sure to at least add one test back in that uses `arma::fmat` as a data type?  This will help us ensure that the template types for data types are working correctly. :+1:

Reply:
Yeah. This one also relies on Boston dataset :)